### DISCUSSION TITLE
Allow private state in Pinia stores

### DISCUSSION THREAD

**Discussion by Koutena** [vuejs/pinia#2789] [Help and Questions] [answered]
As changing state from outside is considered a bad practice, why does Pinia allow accessing and modifying state from outside? Is there a solution to make state private and inaccessible from outside?

I'm using TypeScript and it's crucial for me to preserve its features.

**Example code:**
```ts
// test-store.ts
import { defineStore } from 'pinia';

interface RootState {
  _value: string;
}

export const useTestStore = defineStore({
  id: 'test-store',
  state: (): RootState => ({
    _value: '',
  }),

  actions: {
    updateValue(value: boolean) {
      this._value = value;
    },
  },

  getters: {
    value(): string {
      return this._value;
    },
  }
});
```

**Comment by posva** [comment, marked as answer]
Duplicate of https://github.com/vuejs/pinia/discussions/2756
